ecs-update/ecs-update.go

```diff
@@ -131,18 +138,29 @@ func updatePackage(path, ecsVersion string) error {
 		}
 	}
 
+	if formatVersion != "" {
+		pkg.Manifest.OriginalData.FormatVersion = formatVersion
+		if formatVersion == "2.0.0" {
+			pkg.Manifest.OriginalData.License = ""
+		}
+		err = WriteDocument(&pkg.Manifest, pkg.Manifest.WriteYAML)
```

Comment:
I think there are two problems that stem from gopkg.in/yaml not being able to roundtrip content while preserving the format.

1. Marshaling the manifest.yml back out results in larger than necessary diffs and possibly loss of comments. I introduced some big hacks (which are probably more complicated than alternative approaches) to edit the YAML content for the other changes that are being made. My hack is not sustainable.
\
Probably a better approach would be have a consistent YAML formatting applied to the files such that it does not matter if the tool produces a different ordering since we can just sort it again. But the comments being lost is still a problem. Maybe `yq` commands could be an option.
3. My manifest structs do not have all of the possible attributes. So when you marshal the struct back out you lose attributes. This was OK before when I was only reading the attributes that I needed and was editing the original YAML code and yaml.Node. The solution for this is to use structs generated from package-spec.

Reply:
I've replace the logic here with a real YAML AST handler so those issues should no longer be problems.